test/AxelarGateway.js

```diff
@@ -1401,6 +1877,50 @@ describe('AxelarGateway', () => {
             await deployGateway();
         });
 
+        it('should revert if token does not exist', async () => {
+            const tokenSymbol = getRandomString(10);
+
+            const amount = 1000;
+            const chain = 'Polygon';
+            const destination = '0xb7900E8Ec64A1D1315B6D4017d4b1dcd36E6Ea88';
+            const payload = defaultAbiCoder.encode(['address', 'address'], [owner.address, destination]);
+
+            await expect(gateway.callContractWithToken(chain, destination, payload, tokenSymbol, amount)).to.be.revertedWithCustomError(
+                gateway,
+                'TokenDoesNotExist',
+            );
+        });
+
+        it('should revert if token amount is invalid', async () => {
+            const tokenSymbol = getRandomString(10);
+            const data = buildCommandBatch(
+                await getChainId(),
+                [getRandomID()],
+                ['deployToken'],
+                [getDeployCommand(tokenName, tokenSymbol, decimals, cap, ethers.constants.AddressZero, 0)],
```

Comment:
instead of deploying a new token, you could register an existing test token (that can be deployed exactly once before all tests). deployToken for an existing token is much cheaper

Reply:
I deployed an external token in the before block and used it in a few tests, there were still 2 tests that were having issue using this shared external token, so I kept those tests the same.